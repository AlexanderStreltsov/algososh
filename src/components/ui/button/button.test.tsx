import { type ReactElement } from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Button } from "./button";

const checkWithSnapshot = (element: ReactElement) => {
  const tree = renderer.create(element).toJSON();
  expect(tree).toMatchSnapshot();
};

describe("ui component: Button", () => {
  it("renders correctly with text", () =>
    checkWithSnapshot(<Button text="test" />));

  it("renders correctly without props", () => checkWithSnapshot(<Button />));

  it("renders correctly with disabled state", () =>
    checkWithSnapshot(<Button disabled />));

  it("renders correctly with loader state", () =>
    checkWithSnapshot(<Button isLoader />));

  it("get callback correctly with button click", () => {
    const handleClickMock = jest.fn();
    render(<Button onClick={handleClickMock} />);
    fireEvent.click(screen.getByRole("button"));
    expect(handleClickMock).toHaveBeenCalled();
  });
});
