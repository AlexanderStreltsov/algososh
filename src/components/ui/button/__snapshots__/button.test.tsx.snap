// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui component: Button renders correctly with disabled state 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`ui component: Button renders correctly with loader state 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`ui component: Button renders correctly with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;

exports[`ui component: Button renders correctly without props 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
